pubtools/pulplib/_impl/model/maintenance.py

```diff
@@ -0,0 +1,179 @@
+import logging
+import datetime
+import json
+import os
+
+import jsonschema
+
+from pubtools.pulplib._impl import compat_attr as attr
+from ..schema import load_schema
+from .common import InvalidDataException
+
+
+LOG = logging.getLogger("pubtools.pulplib")
+
+
+USER = os.environ.get("USER")
+HOSTNAME = os.environ.get("HOSTNAME")
+OWNER = "%s@%s" % (USER, HOSTNAME)
+
+
+def iso_time_now():
+    return datetime.datetime.utcnow().strftime("%Y-%m-%dT%H:%M:%SZ")
+
+
+@attr.s(kw_only=True, frozen=True)
+class MaintenanceEntry(object):
+    """Details about the maintenance"""
+
+    id = attr.ib(default=None, type=str)
+    """ID of repository in maintenance"""
```

Comment:
In our current solution, there's actually no guarantee that the repo IDs in the maintenance report match existing repos.  So, this could point at a repository which doesn't exist.

I'm OK with that caveat of the design, but I think it's worthwhile mentioning in our docs somewhere.  Maybe right here is the place?  Something like:

```suggestion
    """ID of repository in maintenance.
    Note: there is no guarantee that a repository of this ID currenly exists in the Pulp server."""
```